ios/brave-ios/Sources/AIChat/Components/AIChatView.swift

```diff
@@ -0,0 +1,555 @@
+          }
+        }
+    })
+  }
+  
+  @ViewBuilder
+  private func apiErrorViews(for error: AiChat.APIError) -> some View {
+    if error == .connectionIssue {
+      AIChatNetworkErrorView() {
+        if !model.conversationHistory.isEmpty {
+          model.retryLastRequest()
+        }
+      }
+        .padding()
+    } else if error == .rateLimitReached {
+      AIChatPremiumUpsellView(
+        upsellType: .rateLimit,
+        upgradeAction: {
+          isPremiumPaywallPresented = true
+        },
+        dismissAction: {
+          if let basicModel = model.models.first(where: { $0.access == .basic }) {
+            model.changeModel(modelKey: basicModel.key)
+            model.retryLastRequest()
+          } else {
+            Logger.module.error("No basic models available")
+          }
+        }
+      )
+      .padding()
+    } else if error == .contextLimitReached {
+      AIChatContextLimitErrorView()
+        .padding()
+    }
+  }
+  
+  private var feedbackView: some View {
+    AIChatFeedbackView(
+      model: AIChatSpeechRecognitionModel(
+        speechRecognizer: speechRecognizer,
+        activeInputView: $voiceSearchActiveInputView,
+        isVoiceEntryPresented: $isVoiceEntryPresented,
+        isNoMicrophonePermissionPresented: $isNoMicrophonePermissionPresented
+      ),
+      onSubmit: { category, feedback in
+        guard let feedbackIndex = customFeedbackIndex else {
+          feedbackToast = .error(message: Strings.AIChat.feedbackSubmittedErrorTitle)
+          return
+        }
+        
+        Task { @MainActor in
+          await model.submitFeedback(
+            category: category,
+            feedback: feedback,
+            ratingId: "\(feedbackIndex)")
+        }
+        
+        customFeedbackIndex = nil
+        feedbackToast = .success(isLiked: true)
+      },
+      onCancel: {
+        customFeedbackIndex = nil
+        feedbackToast = .none
+      },
+      openURL: { url in
+        if url.host == "dismiss" {
+          //TODO: Dismiss feedback learn-more prompt
+        } else {
+          openURL(url)
+        }
+      }
+    )
+    .padding()
+  }
+  
+  private var microphoneAccessAlert: Alert {
+    Alert(
+      title: Text(Strings.AIChat.microphonePermissionAlertTitle),
+      message: Text(Strings.AIChat.microphonePermissionAlertDescription),
+      primaryButton: Alert.Button.default(
+        Text(Strings.settings),
+        action: {
+          let url = URL(string: UIApplication.openSettingsURLString)!
+          UIApplication.shared.open(url, options: [:], completionHandler: nil)
+        }
+      ),
+      secondaryButton: Alert.Button.cancel(Text(Strings.CancelString))
+    )
+  }
+  
+  private func hideKeyboard() {
+    UIApplication.shared.sendAction(
+      #selector(UIResponder.resignFirstResponder),
+      to: nil,
+      from: nil,
+      for: nil
+    )
+  }
+}
+
+@available(iOS 17.0, *)
+#Preview(traits: .sizeThatFitsLayout) {
```

Comment:
Just realized you're missing `#if DEBUG` conditional compilation on all #Preview's, please add those to all Preview's